Profiler/CProfilerCallback.h

```diff
@@ -1,157 +1,184 @@
+public:
+
+	/** Shuts down the profiler from the DllMain function on Dll detach if it is still running. */
+	static void ShutdownFromDllMainDetach();
+
+	/** Constructor. */
+	CProfilerCallback();
+
+	/** Destructor. */
+	virtual ~CProfilerCallback();
+
+	/** Initializer. Called at profiler startup. */
+	STDMETHOD(Initialize)(IUnknown* pICorProfilerInfoUnk);
+
+	/** Write coverage information to log file at shutdown. */
+	STDMETHOD(Shutdown)();
+
+	/** Store information about jitted method. */
+	STDMETHOD(JITCompilationFinished)(FunctionID functionID, HRESULT hrStatus, BOOL fIsSafeToBlock);
+
+	/** Write loaded assembly to log file. */
+	STDMETHOD(AssemblyLoadFinished)(AssemblyID assemblyID, HRESULT hrStatus);
+
+	/** Record inlining of method, but generally allow it. */
+	STDMETHOD(JITInlining)(FunctionID callerID, FunctionID calleeID, BOOL* pfShouldInline);
+
+	/**
+	 * Implements the actual shutdown procedure. Must only be called once.
+	 * If clrIsAvailable is true, also tries to force a GC.
+	 * Note that forcing a GC after the CLR has shut down can result in deadlocks so this
+	 * should be set only when calling from a CLR callback.
+	 */
+	void CProfilerCallback::ShutdownOnce(bool clrIsAvailable);
+
+private:
+	/** Synchronizes profiling callbacks. */
+	CRITICAL_SECTION callbackSynchronization;
+
+	CRITICAL_SECTION methodSetSynchronization;
+
+	/** Default size for arrays. */
+	static const int BUFFER_SIZE = 2048;
+
+	/** Counts the number of assemblies loaded. */
+	int assemblyCounter = 1;
+
+	bool isTestCaseRecording = false;
+	CProfilerCallback* callbackInstance = NULL;
+
+	Config config = Config(WindowsUtils::getConfigValueFromEnvironment);
+
+	/**
+	 * Maps from assembly IDs to assemblyNumbers (determined by assemblyCounter).
+	 * It is used to identify the declaring assembly for functions.
+	 */
+	std::map<AssemblyID, int> assemblyMap;
+
+	/**
+	 * Info object that keeps track of jitted methods.
+	 */
+	std::vector<FunctionInfo> jittedMethods;
+
+	/**
+	 * Keeps track of inlined methods.
+	 * We use the set to efficiently determine if we already noticed an inlined method.
+	 */
+	function_id_set inlinedMethodIds;
+
+	/**
+	 * Keeps track of inlined methods.
+	 * We use the vector to uniquely store the information about inlined methods.
+	 */
+	std::vector<FunctionInfo> inlinedMethods;
+
+	/** Smart pointer to the .NET framework profiler info. */
+	CComQIPtr<ICorProfilerInfo8> profilerInfo;
+
+	/** The log to write all results and messages to. */
+	TraceLog traceLog;
+
+	/** The log to write attach and detatch events to */
+	AttachLog attachLog;
+
+	/** Inter-process connection for TIA communication. null if not in TIA mode. */
+	Ipc* ipc = NULL;
+
+	/** Callback that is being called when a testcase starts. */
+	void onTestStart(std::string testName);
+
+	/** Callback that is being called when a testcase ends. */
+	void onTestEnd(std::string result = "", std::string duration = "");
```

Comment:
As far as I can see `result` and `duration` are read-only
```suggestion
	void onTestEnd(const std::string& result = "", const std::string& duration = "");
```